addons/mail/static/src/model/tests/model_field_command/create_tests.js

```diff
@@ -0,0 +1,125 @@
+
+    const contact = this.env.models['test.contact'].create({ id: 10 });
+    contact.update({ address: create({ id: 20 })});
+    const address = this.env.models['test.address'].findFromIdentifyingData({ id: 20 });
+    assert.strictEqual(
+        contact.address,
+        address,
+        'create: should create and link a record for an empty x2one field'
+    );
+    assert.strictEqual(
+        address.contact,
+        contact,
+        'the inverse relation should be set as well'
+    );
+});
+
+QUnit.test('create: should create and replace a record for a non-empty x2one field', async function (assert) {
+    assert.expect(3);
+    await this.start();
+
+    const contact = this.env.models['test.contact'].create({
+        id: 10,
+        address: create({ id: 10 }),
+    });
+    const address10 = this.env.models['test.address'].findFromIdentifyingData({ id: 10 });
+    contact.update({ address: create({ id: 20 })});
+    const address20 = this.env.models['test.address'].findFromIdentifyingData({ id: 20 });
+    assert.strictEqual(
+        contact.address,
+        address20,
+        'create: should create and replace a record for a non-empty x2one field'
+    );
+    assert.strictEqual(
+        address20.contact,
+        contact,
+        'the inverse relation should be set as well'
+    );
+    assert.strictEqual(
+        address10.contact,
+        undefined,
+        'the original relation should be dropped'
+    );
+});
+
+QUnit.test('create: should create and link a record for an empty x2many field', async function (assert) {
+    assert.expect(2);
+    await this.start();
+
+    const contact = this.env.models['test.contact'].create({ id: 10 });
+    contact.update({ tasks: create({ id: 10 }) });
+    const task = this.env.models['test.task'].findFromIdentifyingData({ id: 10 });
+    assert.ok(
+        contact.tasks instanceof Array &&
+        contact.tasks.length === 1 &&
+        contact.tasks[0] === task,
+        'create: should create and link a record for an empty x2many field'
+    );
+    assert.strictEqual(
+        task.responsible,
+        contact,
+        'the inverse relation should be set as well'
+    );
+});
+
+QUnit.test('create: should create and add a record for a non-empty x2many field', async function (assert) {
+    assert.expect(2);
+    await this.start();
+
+    const contact = this.env.models['test.contact'].create({
+        id: 10,
+        tasks: create({
+            id: 10,
+        }),
+    });
+    const task10 = this.env.models['test.task'].findFromIdentifyingData({ id: 10 });
+    contact.update({ tasks: create({ id: 20 }) });
+    const task20 = this.env.models['test.task'].findFromIdentifyingData({ id: 20 });
+    assert.ok(
+        contact.tasks instanceof Array &&
+        contact.tasks.length === 2 &&
+        contact.tasks.includes(task10) &&
+        contact.tasks.includes(task20),
+        'create: should create and add a record for a non-empty x2many field'
+    );
```

Comment:
Make one assert per condition, that way it's easier to debug if one fails.
This comment applies to all tests.